### PR TITLE
feat: adding towncrier

### LINKED ISSUE #2885: Try using towncrier for changelog
https://towncrier.readthedocs.io/en/stable/

This might be easier to use than manually changing the changelog file. Could be done as part of the documentation gsoc project

### PR DESCRIPTION
**Pull request by arunavabasu-03**
# Description

Using towncrier for CHANGELOG.md 

Fixes #2885

## Type of change

Using towncrier for CHANGELOG.md 

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ ] No style issues: `$ pre-commit run` (or `$ nox -s pre-commit`) (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [ ] All tests pass: `$ python run-tests.py --all` (or `$ nox -s tests`)
- [ ] The documentation builds: `$ python run-tests.py --doctest` (or `$ nox -s doctests`)

You can run integration tests, unit tests, and doctests together at once, using `$ python run-tests.py --quick` (or `$ nox -s quick`).

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

**Comment by arunavabasu-03** [comment]
@tinosulzer pls check do you want this way or their is any approach i have to follow , just done for v23.5 , if it is ok then i goona add all the versions !!

File: changelogs/v23.5/bugfix.rst [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Bug Fixes
+=========
+
+- Realign 'count' increment in CasadiSolver.\_integrate() ([#2986](https://github.com/pybamm-team/PyBaMM/pull/2986))
+- Fix `pybamm_install_odes` and update the required SUNDIALS version ([#2958](https://github.com/pybamm-team/PyBaMM/pull/2958))
```
Comment (Saransh-cpp) [inline changelogs/v23.5/bugfix.rst:5]: [Links are handled differently](https://docutils.sourceforge.io/docs/user/rst/quickref.html#hyperlink-targets) in `rst`, which is why `pre-commit` is failing.

**Review by Saransh-cpp — CHANGES_REQUESTED** [review]
Thanks for working on this, @arunavabasu-03! 🙂 

Took a quick look, and will deep dive into `towncrier` in a few days!

File: changelogs/v23.5/breakingchanges.rst [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Breaking changes
+=========
```
Comment (Saransh-cpp) [inline changelogs/v23.5/breakingchanges.rst:2]:
```suggestion
Breaking changes
================
```

File: changelogs/v23.5/bugfix.rst (hunk 1 above)
Comment (Saransh-cpp) [inline changelogs/v23.5/bugfix.rst:5]: `rst` uses double backticks
```suggestion
- Fix ``pybamm_install_odes`` and update the required SUNDIALS version ([#2958](https://github.com/pybamm-team/PyBaMM/pull/2958))
```

File: changelogs/v23.5/bugfix.rst [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+- Fixed a bug where all data included in a BPX was incorrectly assumed to be given as a function of time.([#2957](https://github.com/pybamm-team/PyBaMM/pull/2957))
+- Remove brew install for Mac from the recommended developer installation options for SUNDIALS ([#2925](https://github.com/pybamm-team/PyBaMM/pull/2925))
+- Fix `bpx.py` to correctly generate parameters for "lumped" thermal model ([#2860](https://github.com/pybamm-team/PyBaMM/issues/2860))
```
Comment (Saransh-cpp) [inline changelogs/v23.5/bugfix.rst:8]:
```suggestion
- Fix ``bpx.py`` to correctly generate parameters for "lumped" thermal model ([#2860](https://github.com/pybamm-team/PyBaMM/issues/2860))
```

File: changelogs/v23.5/feature.rst [hunk 4]
```diff
@@ -0,0 +1,6 @@
+Features
+=========
+
+- Enable multithreading in IDAKLU solver ([#2947](https://github.com/pybamm-team/PyBaMM/pull/2947))
+- If a solution contains cycles and steps, the cycle number and step number are now saved when `solution.save_data()` is called ([#2931](https://github.com/pybamm-team/PyBaMM/pull/2931))
```
Comment (Saransh-cpp) [inline changelogs/v23.5/feature.rst:5]:
```suggestion
- If a solution contains cycles and steps, the cycle number and step number are now saved when ``solution.save_data()`` is called ([#2931](https://github.com/pybamm-team/PyBaMM/pull/2931))
```

File: changelogs/v23.5/feature.rst [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+- Experiments can now be given a `start_time` to define when each step should be triggered ([#2616](https://github.com/pybamm-team/PyBaMM/pull/2616))
```
Comment (Saransh-cpp) [inline changelogs/v23.5/feature.rst:6]:
```suggestion
- Experiments can now be given a ``start_time`` to define when each step should be triggered ([#2616](https://github.com/pybamm-team/PyBaMM/pull/2616))
```

File: changelogs/v23.5/optimizations.rst [hunk 6]
```diff
@@ -0,0 +1,7 @@
+Optimizations
+=========
```
Comment (Saransh-cpp) [inline changelogs/v23.5/optimizations.rst:2]:
```suggestion
Optimizations
=============
```

File: changelogs/v23.5/optimizations.rst [hunk 7]
```diff
@@ -0,0 +1,7 @@
...
+
+- Test `JaxSolver`'s compatibility with Python `3.8`, `3.9`, `3.10`, and `3.11` ([#2958](https://github.com/pybamm-team/PyBaMM/pull/2958))
+- Update Jax (0.4.8) and JaxLib (0.4.7) compatibility ([#2927](https://github.com/pybamm-team/PyBaMM/pull/2927))
+- Migrate from `tox=3.28` to `nox` ([#3005](https://github.com/pybamm-team/PyBaMM/pull/3005))
+- Removed `importlib_metadata` as a required dependency for user installations ([#3050](https://github.com/pybamm-team/PyBaMM/pull/3050))
```
Comment (Saransh-cpp) [inline changelogs/v23.5/optimizations.rst:7]:
```suggestion
- Test ``JaxSolver``'s compatibility with Python ``3.8``, ``3.9``, ``3.10``, and ``3.11`` ([#2958](https://github.com/pybamm-team/PyBaMM/pull/2958))
- Update Jax (0.4.8) and JaxLib (0.4.7) compatibility ([#2927](https://github.com/pybamm-team/PyBaMM/pull/2927))
- Migrate from ``tox=3.28`` to ``nox`` ([#3005](https://github.com/pybamm-team/PyBaMM/pull/3005))
- Removed ``importlib_metadata`` as a required dependency for user installations ([#3050](https://github.com/pybamm-team/PyBaMM/pull/3050))
```

File: changelogs/v23.5/bugfix.rst (hunk 1 above)
Comment (Saransh-cpp) [inline changelogs/v23.5/bugfix.rst:5]:
```suggestion
- Fix ``pybamm_install_odes`` and update the required SUNDIALS version ([#2958](https://github.com/pybamm-team/PyBaMM/pull/2958))
```

**Review by Saransh-cpp — CHANGES_REQUESTED** [review]

**Comment by agriyakhetarpal** [comment]
I think we can convert the entire `CHANGELOG.md` file using [executablebooks/rst-to-myst](https://rst-to-myst.readthedocs.io/en/stable/index.html), or we could just [keep it in Markdown](https://towncrier.readthedocs.io/en/stable/markdown.html) like it is already because I feel the syntax is more intuitive

**Comment by agriyakhetarpal** [comment]
@arunavabasu-03, was this in error or would you not be working on this feature anymore?

**Pull request closed without merging**